From 08515a8469b89f91479e882fde4c8913d9072517 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Tue, 28 May 2024 04:14:15 +0530
Subject: [PATCH 63/98] msdk: Fix libdrm dependency detection and usage

drm_fourcc.h should be picked up via the pkgconfig include, not the
system includedir directly.

Also consolidate the libdrm usage in va and msdk.

All this allows it to be picked up consistently (via the subproject,
for example).

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/6932>
---
 gst-libs/gst/va/meson.build       |  2 --
 meson.build                       | 11 +++++++++++
 sys/msdk/gstmsdkallocator_libva.c |  2 +-
 sys/msdk/gstmsdkcaps.c            |  2 +-
 sys/msdk/gstmsdkdec.c             |  2 +-
 sys/msdk/gstmsdkenc.c             |  2 +-
 sys/msdk/gstmsdkvpp.c             |  2 +-
 sys/msdk/gstmsdkvpputil.c         |  2 +-
 sys/msdk/meson.build              |  1 -
 9 files changed, 17 insertions(+), 9 deletions(-)

diff --git a/gst-libs/gst/va/meson.build b/gst-libs/gst/va/meson.build
index b9a8ae7066c6..d83b9e3f3fec 100644
--- a/gst-libs/gst/va/meson.build
+++ b/gst-libs/gst/va/meson.build
@@ -81,10 +81,8 @@ if host_system == 'linux'
   endif
   platform_deps += [libva_drm_dep]
 
-  libdrm_dep = dependency('libdrm', version: '>=2.4', required: get_option('drm'))
   if libdrm_dep.found()
     platform_deps += [libdrm_dep]
-    extra_args += ['-DHAVE_LIBDRM']
   endif
 
   va_sources += va_linux_sources
diff --git a/meson.build b/meson.build
index 6707dcec5d10..a9feb88fb357 100644
--- a/meson.build
+++ b/meson.build
@@ -398,6 +398,17 @@ if x11_dep.found()
   cdata.set('HAVE_X11', 1)
 endif
 
+# Optional dep of msdk and va
+if host_system not in ['darwin', 'ios', 'android', 'windows']
+  libdrm_dep = dependency('libdrm', version : '>=2.4.50',
+    required : get_option('drm').enabled() or get_option('msdk').enabled(),
+  )
+  cdata.set('HAVE_LIBDRM', libdrm_dep.found())
+else
+  libdrm_dep = dependency('', required: false)
+endif
+
+
 #
 # Solaris and Illumos distros split a lot of networking-related code
 # into '-lsocket -lnsl'.  Anything that calls socketpair(), getifaddr(),
diff --git a/sys/msdk/gstmsdkallocator_libva.c b/sys/msdk/gstmsdkallocator_libva.c
index e92719af76c0..c20b8c068dd7 100644
--- a/sys/msdk/gstmsdkallocator_libva.c
+++ b/sys/msdk/gstmsdkallocator_libva.c
@@ -32,7 +32,7 @@
 
 #include <va/va.h>
 #include <va/va_drmcommon.h>
-#include <libdrm/drm_fourcc.h>
+#include <drm_fourcc.h>
 #include <unistd.h>
 #include "gstmsdkallocator.h"
 #include "gstmsdkallocator_libva.h"
diff --git a/sys/msdk/gstmsdkcaps.c b/sys/msdk/gstmsdkcaps.c
index 80c353bf6dfd..a10f43eba595 100644
--- a/sys/msdk/gstmsdkcaps.c
+++ b/sys/msdk/gstmsdkcaps.c
@@ -32,7 +32,7 @@
 #include "gstmsdkcaps.h"
 
 #ifndef _WIN32
-#include <libdrm/drm_fourcc.h>
+#include <drm_fourcc.h>
 #include <gst/video/video-info-dma.h>
 #include "gstmsdkallocator_libva.h"
 #include <gst/va/gstvavideoformat.h>
diff --git a/sys/msdk/gstmsdkdec.c b/sys/msdk/gstmsdkdec.c
index c7304e2e7a3d..11122914c89b 100644
--- a/sys/msdk/gstmsdkdec.c
+++ b/sys/msdk/gstmsdkdec.c
@@ -41,7 +41,7 @@
 #include "gstmsdkcaps.h"
 
 #ifndef _WIN32
-#include <libdrm/drm_fourcc.h>
+#include <drm_fourcc.h>
 #include <gst/va/gstvaallocator.h>
 #include <gst/va/gstvavideoformat.h>
 #else
diff --git a/sys/msdk/gstmsdkenc.c b/sys/msdk/gstmsdkenc.c
index f000aaf12c0a..76b393d26597 100644
--- a/sys/msdk/gstmsdkenc.c
+++ b/sys/msdk/gstmsdkenc.c
@@ -51,7 +51,7 @@
 #include "gstmsdkcaps.h"
 
 #ifndef _WIN32
-#include <libdrm/drm_fourcc.h>
+#include <drm_fourcc.h>
 #include "gstmsdkallocator_libva.h"
 #include <gst/va/gstvaallocator.h>
 #else
diff --git a/sys/msdk/gstmsdkvpp.c b/sys/msdk/gstmsdkvpp.c
index b25178d134c7..8eebdbf53139 100644
--- a/sys/msdk/gstmsdkvpp.c
+++ b/sys/msdk/gstmsdkvpp.c
@@ -60,7 +60,7 @@
 #include "gstmsdkallocator.h"
 
 #ifndef _WIN32
-#include <libdrm/drm_fourcc.h>
+#include <drm_fourcc.h>
 #include "gstmsdkallocator_libva.h"
 #include <gst/va/gstvaallocator.h>
 #include <gst/va/gstvavideoformat.h>
diff --git a/sys/msdk/gstmsdkvpputil.c b/sys/msdk/gstmsdkvpputil.c
index 8324bb18f1f2..bbd14790fe60 100644
--- a/sys/msdk/gstmsdkvpputil.c
+++ b/sys/msdk/gstmsdkvpputil.c
@@ -30,7 +30,7 @@
 #include "gstmsdkcaps.h"
 
 #ifndef _WIN32
-#include <libdrm/drm_fourcc.h>
+#include <drm_fourcc.h>
 #include <gst/va/gstvavideoformat.h>
 #else
 #include <gst/d3d11/gstd3d11.h>
diff --git a/sys/msdk/meson.build b/sys/msdk/meson.build
index 51bf953a0e55..b73370cda129 100644
--- a/sys/msdk/meson.build
+++ b/sys/msdk/meson.build
@@ -184,7 +184,6 @@ if host_machine.system() == 'windows'
 else
   libdl_dep = cc.find_library('dl', required: get_option('msdk'))
   libgudev_dep = dependency('gudev-1.0', required: get_option('msdk'), allow_fallback: true)
-  libdrm_dep = dependency('libdrm', required: get_option('msdk'))
   msdk_deps = declare_dependency(dependencies: [gstva_dep, libdl_dep, libgudev_dep, libdrm_dep])
   msdk_deps_found = gstva_dep.found() and libdl_dep.found() and libgudev_dep.found() and libdrm_dep.found()
 endif
-- 
2.43.0

